pkg/state/state_run.go

```diff
@@ -165,21 +165,6 @@ func GroupReleasesByDependency(releases []Release, opts PlanOptions) ([][]Releas
 		}
 	}
 
-	for _, r := range releases {
-		if !r.Filtered {
-			for _, n := range r.Needs {
-				// To map n into idToReleases correctly, prepend KubeContext to n.
-				if r.KubeContext != "" {
-					n = r.KubeContext + "/" + n
-				}
-				if _, ok := idToReleases[n]; !ok {
-					id := ReleaseToID(&r.ReleaseSpec)
-					return nil, fmt.Errorf("%q depends on nonexistent release %q", id, n)
-				}
-			}
-		}
-	}
```

Comment:
See L209-219- One of the intentions of #1772 was to delegate this duplication detection to the `variantdev/dag`. This is now redundant. I've updated a few test cases for missing dependency errors accordingly.